src/coreComponents/common/TypeDispatch.hpp

```diff
@@ -0,0 +1,341 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file TypeDispatch.hpp
+ *
+ * Collection of tools to help dispatch templated functions on types
+ */
+
+#ifndef GEOSX_COMMON_TYPEDISPATCH_HPP
+#define GEOSX_COMMON_TYPEDISPATCH_HPP
+
+#include "common/DataTypes.hpp"
+
+#include <camp/camp.hpp>
```

Comment:
TPL's should be with `<>`. AFAIK the only real difference is that warnings are silenced for them.

Reply:
We seem to have a mix of both styles sprinkled around various parts of the code (even within the same component, e.g. `linearAlgebra`). If we were to standardize/cleanup these, I would vote for `<>` because it searches in `-I` directories only which seems appropriate for TPLs, whereas `""` also searches in current and parent directories.
https://gcc.gnu.org/onlinedocs/cpp/Include-Syntax.html

Reply:
The one thing is that sometimes it takes a bit of manipulation to force a TPL added from CMake's `find_package` to treat it's includes as system. I think it's worth the effort.